aten/src/ATen/native/cuda/ReduceSumProdKernel.cu

```diff
@@ -13,6 +13,21 @@ namespace at::native {
 template <typename scalar_t, typename acc_t = scalar_t, typename out_t = scalar_t>
 struct sum_functor {
   void operator()(TensorIterator& iter) {
+    #if (defined USE_ROCM) && (PYTORCH_ROCM_ARCH==gfx942)
```

Comment:
This check PYTORCH_ROCM_ARCH ==gfx942 will not work.

Reply:
good catch. Let me fix it.

Reply:
yeah, lets remove this check all together if MI250X perf is decent with the fix

Reply:
I left USE_ROCM, but let me know if we need to remove it as well.